### ISSUE TITLE
Enemy unit re-skin mods compabilitys

### ISSUE THREAD

**Issue by sirben99** [segabl/pd2-streamlined-heisting#13]
Hey!

I was wondering are enemy unit overhauls how compatible with Streamlined Heisting?
I remember having crashes along time ago when trying enemy re-skin mods with this one, the game would crash or the shotgun guys would have the normal helmet instead of glass one. You mentioned that you would propably make a seperate file for those kind of mods in the mod page some time ago, and i was wondering if this is still the case or there arent any official plans to make these kinds of mods compatible?

E.g of one of these mods: https://modworkshop.net/mod/31196#description

This mod accually added compability with Streamlined: https://modworkshop.net/mod/24487

Sorry if i explained this very confusingly, if you have any further questions of what i mean let me know!

Ben

**Comment by segabl** [comment]
The crashing issues come from reskins that expect some original parts of the models (material config, object/unit files) where in some cases I change those in streamlined heisting. for compatibility these reskins would have to include the vanilla assets of whatever other connected files sh changes, which is obviously not something you would do for a reskin (you'd expect users to use reskins over vanilla assets). The issue here with making the visual changes a separate mod is that I would have to double override a lot of assets in case of using both, because some changes to related non model assets are still required for non visual stuff.

TL;DR asset overrides are messy but usually work if they are made as superblt mod instead of mod_overrides

**Comment by sirben99** [comment]
Ah, yeah the seperate file would take alot of work, and as you mentioned (And i hope i got it right), some changes are not purpely visual and would need extra tweaking, which might not be worth the effort since there arent many re-skin mods these days, and many of them arent updated in years.

Thank you for the answer and detailed explonation!

**Comment by segabl** [comment]
On the plus side, reskins made with superBLTs asset loading are very simple to be made compatible since it's just a matter of using the right mod priority

**Comment by sirben99** [comment]
Hmm, how easy would it be for me (Inexperienced modder) to change Mod_overrides to SuperBLT? For e.g Prettier FBI Units: https://modworkshop.net/mod/31196#description
Or do i need the authors permission for editing his mod for personal use?

**Comment by segabl** [comment]
you'd have to move the mod to your mods folder and make a mod.txt and supermod.xml similar to what this mod (or actually so much win enemies) does. no need for any code or stuff, just the assets being defined in the supermod.xml

**Comment by sirben99** [comment]
Thank you for the answer! I accually got the BLT hook working for mod_overrides mod, but im having a bit of an issue with this mod: https://modworkshop.net/mod/354#description
Ik its 5 years old, but it works well besides a few things. For e.g in the screenshots FBI Shotgun dode has FBI written in his arm, but ingame its a FBI logo. 
Also, i noticed that ingame that helmets_heavy_taser_atlas_df.texture file causes CITY SWAT to have green helmets, instead of black ones like in the screenshots.

My question is: Instead of some textures being shared in the shared_textures folder, would it be possible to have texture for one type of enemy only? For e.g: ene_city_swat_1 would only have helmet texture change instead of them all.

Sorry if i explained this poorly, i plan to use different mods for Gensec (CitySwat) anyway, but this cought my eye why some of the textures arent working, even in mod_overrides folder (Unless the author removed them and the screenshots are outdated?).

Do you know what program i should use to edit the textures? For e.g: helmets_atlas_df.texture?

**Comment by segabl** [comment]
the issue here is that these enemies probably rely on the default material configs/object files and most likely those have been changed by streamlined heisting. look at the corresponding assets in streamlined heisting and see if there are material config or other files for that specific unit. if there are, you'll have to copy the original ones from the game and add them additionally
